packages/desktop-gui/src/settings/configuration.jsx

```diff
@@ -48,7 +48,27 @@ const normalizeWithoutMeta = (value = {}) => {
 }
 
 const ObjectLabel = ({ name, data, expanded, from, isNonenumerable }) => {
-  const formattedData = formatValue(data)
+  let formattedData = formatValue(data)
+
+  const iconStyles = {
+    marginLeft: '8px',
+  }
+
+  const editableInputStyles = {
+    height: '18px',
+    width: '100px',
+  }
+  const [isEditable, setIsEditable] = useState(false)
+
+  const [value, setValue] = useState('')
+
+  const [configData, setConfigData] = useState(formattedData)
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    setIsEditable(!isEditable)
+    setConfigData(value)
```

Comment:
I'm able to change config values here, but it only takes effect in the UI labels and not in the app's config state. I'm assuming I need to change it before it gets passed here `<ConfigDisplay data={project.resolvedConfig} />`? 

Can you(or somebody on the team) onboard/ guide me on how to use `hooks` to change the `observer`?